packages/overture-schema-core/src/overture/schema/core/geometry.py

```diff
@@ -0,0 +1,368 @@
+            return {
+                "oneOf": allowed_schemas,
+            }
+
+
+_ALL_GEOMETRY_ALLOWED = GeometryTypeConstraint(*_GEOMETRY_TYPES)
+
+
+class Geometry:
+    geom: BaseGeometry
+
+    def __init__(self, geom: BaseGeometry):
+        self.geom = geom
+
+    def __eq__(self, other: Any) -> bool:
+        return isinstance(other, Geometry) and self.geom == other.geom
+
+    def __hash__(self) -> int:
+        return hash(self.geom)
+
+    def __repr__(self) -> str:
+        return f"<{repr(self.geom)}>"
+
+    def __str__(self) -> str:
+        return self.wkt
+
+    @property
+    def wkt(self) -> str:
+        return self.geom.wkt
+
+    def to_geo_json(self) -> dict[str, Any]:
+        return mapping(self.geom)
+
+    @classmethod
+    def from_geo_json(cls, value: Any) -> "Geometry":
+        # If it's already a Shapely geometry, use it directly
+        if isinstance(value, BaseGeometry):
+            return cls(value)
+
+        if not isinstance(value, dict):
+            raise TypeError(
+                f"value must be a dict; but {repr(value)} has type {type(value).__name__}"
+            )
+
+        type_ = value.get("type")
+
+        if type_ not in _GEOMETRY_TYPES:
+            raise ValueError(
+                f"allowed_types contains invalid value {repr(type_)} (allowed: {_GEOMETRY_TYPES})"
+            )
+
+        return cls(shape(value))
+
+    @classmethod
+    def from_wkb(cls, value: bytes) -> "Geometry":
+        """Create a Geometry from Well-Known Binary (WKB) bytes."""
+        if not isinstance(value, bytes):
+            raise TypeError(
+                f"value must be bytes; but {repr(value)} has type {type(value).__name__}"
+            )
+
+        try:
+            geom = wkb.loads(value)
+            return cls(geom)
+        except Exception as e:
+            raise ValueError(f"invalid WKB data: {str(e)}") from e
+
+    @classmethod
+    def from_wkt(cls, value: str) -> "Geometry":
+        """Create a Geometry from Well-Known Text (WKT) string."""
+        if not isinstance(value, str):
+            raise TypeError(
+                f"value must be str; but {repr(value)} has type {type(value).__name__}"
+            )
+
+        try:
+            geom = wkt.loads(value)
+            return cls(geom)
+        except Exception as e:
+            raise ValueError(f"invalid WKT string: {str(e)}") from e
+
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls, _source_type: Any, _handler: GetCoreSchemaHandler
+    ) -> core_schema.CoreSchema:
+        def validator(value: Any, info: ValidationInfo) -> Geometry:
+            try:
+                # Handle Shapely geometry directly
+                if isinstance(value, BaseGeometry):
+                    return cls(value)
+                # Handle GeoJSON dict
+                elif isinstance(value, dict):
+                    return cls.from_geo_json(value)
```

Comment:
This logic being separated here seems to support my view that `from_geo_json` probably shouldn't handle Shapely internally.